libcodechecker/libhandlers/store.py

```diff
@@ -310,124 +292,31 @@ def main(args):
-
-    original_cwd = os.getcwd()
-    empty_metadata = {'working_directory': original_cwd}
-
-    check_commands = []
-    check_durations = []
-    skip_handlers = []
-    items = []
-
-    server_data = (args.host, args.port)
-
-    for input_path in args.input:
-        input_path = os.path.abspath(input_path)
-
-        # WARN: analyze command should create this file!
-        cmp_db = os.path.join(input_path, 'compile_cmd.json')
-        compile_commands = process_compile_db(cmp_db)
-
-        LOG.debug("Parsing input argument: '" + input_path + "'")
-
-        if os.path.isfile(input_path):
-            if not input_path.endswith(".plist"):
-                continue
-
-            items.append((input_path,
-                          context,
-                          empty_metadata,
-                          compile_commands,
-                          server_data))
-
-        elif os.path.isdir(input_path):
-            metadata_file = os.path.join(input_path, "metadata.json")
-            if os.path.exists(metadata_file):
-                with open(metadata_file, 'r') as metadata:
-                    metadata_dict = json.load(metadata)
-                    LOG.debug(metadata_dict)
-
-                    if 'command' in metadata_dict:
-                        check_commands.append(metadata_dict['command'])
-                    if 'timestamps' in metadata_dict:
-                        check_durations.append(
-                            float(metadata_dict['timestamps']['end'] -
-                                  metadata_dict['timestamps']['begin']))
-                    if 'skip_data' in metadata_dict:
-                        # Save previously stored skip data for sending to the
-                        # database, to ensure skipped headers are actually
-                        # skipped --- 'analyze' can't do this.
-                        handle, path = tempfile.mkstemp()
-                        with os.fdopen(handle, 'w') as tmpf:
-                            tmpf.write('\n'.join(metadata_dict['skip_data']))
-
-                        skip_handlers.append(
-                            skiplist_handler.SkipListHandler(path))
-                        os.remove(path)
-            else:
-                metadata_dict = empty_metadata
-
-            _, _, files = next(os.walk(input_path), ([], [], []))
-            for f in files:
-                if not f.endswith(".plist"):
-                    continue
-
-                items.append((os.path.join(input_path, f),
-                              context, metadata_dict, compile_commands,
-                              server_data))
-
-    if len(check_commands) == 0:
-        command = ' '.join(sys.argv)
-    elif len(check_commands) == 1:
-        command = ' '.join(check_commands[0])
-    else:
-        command = "multiple analyze calls: " +\
-                  '; '.join([' '.join(com) for com in check_commands])
+    workspace = util.get_default_workspace()
```

Comment:
Right now, yes, because `--workspace X` is a "symbolic link" for `--config-dir X --sqlite X/codechecker.sqlite`. We already have authentication-related things **of the client** in the user's home folder. All I'm saying is that this `get_default_workspace()` function is supposedly will be removed later.